src/openrct2/management/Marketing.cpp

```diff
@@ -237,3 +237,20 @@ void marketing_new_campaign(const MarketingCampaign& campaign)
         gMarketingCampaigns.push_back(campaign);
     }
 }
+
+void marketing_cancel_campaigns_for_ride(ride_id_t rideId)
+{
+    for (auto it = gMarketingCampaigns.begin(); it != gMarketingCampaigns.end();)
```

Comment:
I think it would be less error-prone to use the [erase-remove](https://en.wikipedia.org/wiki/Erase%E2%80%93remove_idiom) idiom:

```
auto isAdvertisingFn = [rideId](MarketingCampaign& campaign)
{
    if (campaign.Type == ADVERTISING_CAMPAIGN_RIDE_FREE || campaign.Type == ADVERTISING_CAMPAIGN_RIDE)
    {
        return campaign.RideId == rideId;
    }
    return false;
};

auto& v = gMarketingCampaign;
auto removedIt = std::remove_if(v.begin(), v.end(), isAdvertisingFn);
v.erase(removedIt, v.end());
```

Reply:
Agreed, that does look a bit cleaner than my version 👍